### PR TITLE
Feat: User profile

### PR DESCRIPTION
**Pull request by bharati-21**
This PR contains the following.
- Logged in users can view their profile by navigating to profile page using the sidebar (footer in small screens) link
- Logged in users can edit their profile details by clicking on the "Edit Profile" button in the profile page

### CODE REVIEW

**Review by aniketxparihar — APPROVED** [review]
- Clean Code
- Good Folder Structure
- Liked how you are uploading images to cloudinary
- Good use of toastify
- Formatting is a little Off
Overall Good to merge

File: src/components/EditProfileForm/index.jsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+			return;
+		}
+		const url =
+			"https://api.cloudinary.com/v1_1/" +
+			process.env.REACT_APP_CLOUD_NAME +
+			"/image/upload";
+
+		const formData = new FormData();
+		formData.append("file", imageFile);
+		formData.append("upload_preset", "l4ggled4");
```
Comment (rohanmathur91) [inline src/components/EditProfileForm/index.jsx:45]: put this in `.env` and then read, else anyone will be able to upload images in your cloud

File: src/features/Modal/index.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+				<div
+					className="rounded-lg shadow-xl transition-all align-middle max-w-lg h-max w-full relative"
+					ref={modalContainerRef}
+				>
+					{modalChildren === "POST_MODAL" ? (
+						<PostContainer container="modal" />
+					) : modalChildren === "EDIT_PROFILE" ? (
+						<EditProfileForm />
+					) : null}
+				</div>
```
Comment (rohanmathur91) [inline src/features/Modal/index.jsx:54]: nice way of handling multiple modals

**Review by rohanmathur91 — APPROVED** [review]
Code is clean and overall looks good to merge.

File: src/components/EditProfileForm/index.jsx (hunk 1 above)
Comment (bharati-21) [reply src/components/EditProfileForm/index.jsx:45]: Yes, thank you. This will be reflected in the next PR

**Comment by bharati-21** [comment]
> Code is clean and overall looks good to merge.

Thank you for review!

**Comment by bharati-21** [comment]
> * Clean Code
> * Good Folder Structure
> * Liked how you are uploading images to cloudinary
> * Good use of toastify
> * Formatting is a little Off
>   Overall Good to merge

Thank you! Yes, I'm using prettier but not sure why it is a little off on github. 😅

**Pull request merged**